compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/Transforms/PassDetail.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/IR/Iterators.h"
+#include "mlir/Pass/Pass.h"
+
+#define DEBUG_TYPE "iree-amdaie-bufferize-to-allocation"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+static LogicalResult applyBufferizeToAllocation(RewriterBase &rewriter,
+                                                Operation *op,
+                                                Attribute memorySpace) {
+  linalg::BufferizeToAllocationOptions options;
+  options.memcpyOp =
+      linalg::BufferizeToAllocationOptions::MemcpyOp::MaterializeInDestination;
+  options.allocOp = linalg::BufferizeToAllocationOptions::AllocOp::MemrefAlloc;
+  options.bufferizeDestinationOnly = true;
+  options.emitDealloc = true;
+
+  // Bufferize ops.
+  Value buffer =
+      linalg::bufferizeToAllocation(rewriter, options, op, memorySpace);
+  if (!buffer) {
+    LLVM_DEBUG(llvm::dbgs() << "----- failed to bufferize operation -----\n");
+    return failure();
+  }
+  return success();
+}
+
+// TODO(avarma): This is a temporary workaround until we have PaddingStrategy
```

Comment:
nit: if you used an enum class for this it'd make the code easier to understand. Is there a reason using ints is easier?

Reply:
We use int values so that it has good correspondence with the paddingLevel. I've modified this part to make it clear to read.